azure-pipelines.yml

```diff
@@ -159,6 +160,18 @@ jobs:
   condition: and(succeeded(), eq(dependencies.lint.outputs['DetermineBuildType.onlyDocChanges'], '0'))
   pool:
     vmImage: ubuntu-16.04
+
+  strategy:
+    matrix:
+      gcc:
+        meson_init_args: ""
+        artifact_name: sw_build_gcc
+        allowed_to_fail: false
+      clang:
+        meson_init_args: "-t ${TOOLCHAIN_PATH}/meson-riscv32-unknown-elf-clang.txt"
+        artifact_name: sw_build_clang
+        allowed_to_fail: true
```

Comment:
All our Azure Pipelines variables are all-caps currently, let's do that here as well.